### ISSUE TITLE
We should use Docker so it's seamless to use

### ISSUE THREAD

**Issue by westoque** [github/licensed#224]
Trying to see what people think about having a Docker workflow so we can use `licensed` seamlessly. We currently use [brakeman](https://github.com/presidentbeef/brakeman/) in this fashion and would also be good to do for licensed.

```
docker run -v "$(pwd)":/code github/licensed licensed status
```

Thoughts?

**Comment by jonabc** [comment]
@westoque :wave: thats a really great question and I've already spent a long time trying to figure out 😞 .

Due to quite a bit of complexity I ended up dropping any docker-specific work in favor of distributing self-contained executables that can be downloaded for each licensed release.

**TL;DR this is a hard problem for licensed since it has unique environment requirements per usage.  I recommend for users to create custom Dockerfiles for their projects and to install licensed into those images.  There are additional options for some non-docker scenarios**

There's a few major problems that make a tool like licensed not a great idea for "officially supported" docker images.

First, licensed needs an available runtime for each tool/language/framework being used.  This becomes a problem because everyone's use case is different - I've seen projects using go, ruby and npm all in the same repo.  Do users want to run each language in separate docker containers, or in a single container that has all available frameworks?  Do the maintainers of the project need to support a docker image for every combination of tools that licensed supports, for each supported version of each tool/framework?  That sounds... less than fun 😆 

There's an additional issue with an all-in-one image, in that it will be LARGE.  When I attempted to work through this, a workable Haskell installation alone was multiple GB.  Downloading that into CI environments, or locally, sounds terrible and would need to be redone whenever licensed is released - on average maybe once or twice a month.

I also attempted to provide a docker image that contained ruby, licensed and scripting to install each tool as needed in the docker image.  The problem there is that it can add a significant amount of setup time in the container before licensed is usable.

For projects that include ruby, that's easy enough with `gem install licensed [version]` or including licensed in a Gemfile / gemspec and picking it up as part of `bundle install`.

For projects that don't include ruby, every release since [1.2.0](https://github.com/github/licensed/releases/tag/1.2.0) includes executables that can be downloaded and run.  They're not the smallest files as they're basically a self-extracting archive of a ruby runtime and the licensed code but they should be small enough to be alright in most situations.

Recently, some additional tools have been made to make using licensed easier in specific scenarios

1. a bundler plugin - [ref](https://github.com/github/licensed#bundler)
2. a pair of GitHub Actions to help make running licensed in your GitHub Action workflow easier - [ref](https://github.com/github/licensed#github-actions)
